mllib/src/main/scala/org/apache/spark/mllib/util/MLUtils.scala

```diff
@@ -82,6 +82,17 @@ object MLUtils {
           val value = indexAndValue(1).toDouble
           (index, value)
         }.unzip
+
+        // check if indices are one-based and in ascending order
+        var previous = -1
+        var i = 0
+        val indicesLength = indices.size
```

Comment:
nit: `indices.length` (`size` introduces one extra method call, and scala compiler doesn't  optimize it.)